### PR TITLE
Feat/ Added socketPath as a prop for NovuProvider

### LINKED ISSUE #2994: Support for sockets in subdirectories 🚀 Feature:
### 🔖 Feature description

Currently socket connections that are in subdirectories are no properly initialised in `useInitializeSocket` function. 
This means that the socket can only be located in the basepath 

### 🎤 Why is this feature needed ?

If the socket location is anything but `/` it's impossible to connect to a socket with NovuProvider

### ✌️ How do you aim to achieve this?

A new prop called `socketPath` that can be passed together with `socketUrl`. By default this path is `/` but in case the socket URL contains a subdirectory then users have the option to add the subdirectory there. Then this `socketPath` can be passed in the `io` init function as a path option.

### 🔄️ Additional Information

_No response_

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Contributing Guidelines](https://github.com/novuhq/novu/blob/main/CONTRIBUTING.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by Toms-Uluks**
### What change does this PR introduce?

Implements the feature described in https://github.com/novuhq/novu/issues/2994 (Support for sockets in subdirectory)

### Why was this change needed?

Closes https://github.com/novuhq/novu/issues/2994

### CODE REVIEW

**Comment by Cliftonz** [comment]
@Toms-Uluks  Thanks for submitting this PR! 
Looks like there is a few styling errors in the docs and on the widget that the the CI found. 

Once we get those fixed, we can take a deeper look otherwise this looks great. Thanks again!

**Comment by LetItRock** [comment]
hey @Toms-Uluks! 👋 
The PR looks good to me, please also verify if those new props work for the Angular and Vue packages.
Let me know after you check that and fix the CI issues, I'll review it again.

**Comment by Toms-Uluks** [comment]
> @Toms-Uluks Thanks for submitting this PR! Looks like there is a few styling errors in the docs and on the widget that the the CI found.
> 
> Once we get those fixed, we can take a deeper look otherwise this looks great. Thanks again!

Oh sorry about that, should have just used a code block there.

**Comment by Toms-Uluks** [comment]
> hey @Toms-Uluks! 👋 The PR looks good to me, please also verify if those new props work for the Angular and Vue packages. Let me know after you check that and fix the CI issues, I'll review it again.

Hi, yeah it works for all frameworks the same way. And should be good to go for another review now with the fix for the documentation page

**Comment by LetItRock** [comment]
hey @Toms-Uluks, could you fix the linter issues in the widget? thanks! 🙌 https://github.com/novuhq/novu/actions/runs/4413366400/jobs/7745692587

**Review by scopsy — APPROVED** [review]

**Comment by scopsy** [comment]
@Toms-Uluks seems like the tests are failing here due to broken behaviour :( Widget tests are testing the socket interactions from the client perspective

**Comment by Toms-Uluks** [comment]
> @Toms-Uluks seems like the tests are failing here due to broken behaviour :( Widget tests are testing the socket interactions from the client perspective

Hi @scopsy 
Sorry for the delay on this PR, I pulled the head of the branch but I'm getting a lot of errors from DeepSource tests around parts that I haven't touched. I fixed some of them in my lates commit but I would really like to avoid touching larger ones like circular dependencies and  assigning `this` to a local variable.

File: docs/docs/notification-center/react/react-components.md [hunk 1]
```diff
@@ -543,3 +545,4 @@ If you want to hide it, you can use the prop `showUserPreferences` on the `Popov
 :::note
 Facing issues in using notification center? Check out FAQs [here](../FAQ)
 :::
+
```
Comment (jainpawan21) [inline docs/docs/notification-center/react/react-components.md:548]: @Toms-Uluks Please remove this extra line
lint for docs is failing due to this!

**Review by jainpawan21 — CHANGES_REQUESTED** [review]

**Comment by scopsy** [comment]
Hi @Toms-Uluks, I don't have the permissions to merge latest changes to here from next so I can't merge it :( 

Please let me know if you have sometime for this or should we close the PR.

**Comment by scopsy** [comment]
Closing for now as stale 🙏

**Comment by casyalex** [comment]
This is useful when ws host on subdirectory...

**Pull request closed without merging**